### PR TITLE
[TS SDK ] run ts sdk indexer tests in a separate CI job

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
indexer service can be flaky and we dont want those tests to be land blocking for any PR on the aptos repo. This is why we run indexer tests separate from the main sdk test run
### Test Plan
tests are passing

### CODE REVIEW

**Review by banool — APPROVED** [review]

File: .github/workflows/ts-sdk-indexer-tests.yaml
Comment (rustielin) [inline .github/workflows/ts-sdk-indexer-tests.yaml:1]: would it make sense to run this in the same workflow as `sdk-release.yaml`, but just as a different job?

**Review by rustielin — APPROVED** [review]
stamping to unblock, but technically this can be in the same workflow iiuc

File: .github/workflows/sdk-release.yaml [hunk 1]
```diff
@@ -141,3 +141,28 @@ jobs:
...
+
+  # Run the TS SDK indexer tests. Note: indexer service can be flaky and we
+  # dont want those tests to be land blocking for any PR on the aptos repo.
+  # This is why we run those tests separate from
+  # test-sdk-confirm-client-generated-publish.
+  run-indexer-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
```
Comment (rustielin) [inline .github/workflows/sdk-release.yaml:153]: can you pin the version of these third party action deps like in the other job? e.g.
```
actions/setup-node@969bd2663942d722d85b6a8626225850c2f7be4b # pin@v3
```

Comment (0xmaayan) [reply .github/workflows/sdk-release.yaml:153]: should I do the same for checkout?
```
actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3
```

Comment (rustielin) [reply .github/workflows/sdk-release.yaml:153]: yep, and `pnpm/action-setup@v2` as well

Comment (rustielin) [reply .github/workflows/sdk-release.yaml:153]:
```
pnpm/action-setup@537643d491d20c2712d11533497cb47b2d0eb9d5 # pin https://github.com/pnpm/action-setup/releases/tag/v2.2.3
```

Comment (0xmaayan) [reply .github/workflows/sdk-release.yaml:153]: we use it like this in the other job thought https://github.com/aptos-labs/aptos-core/pull/8454/files#diff-f9bdc9424e6b94fc4cdf8b6974e1c90d62ffc04c88d91c80c721022b183df298R62-R64

Comment (rustielin) [reply .github/workflows/sdk-release.yaml:153]: ah that should best be changed as well. pinning the third party dep by SHA is a best practice

**Pull request merged**